homeassistant/components/min_max/sensor.py

```diff
@@ -287,68 +281,89 @@ def extra_state_attributes(self) -> dict[str, Any] | None:
 
     @callback
     def _async_min_max_sensor_state_listener(
-        self, event: EventType, update_state: bool = True
+        self, event: EventType | None, update_state: bool = True
     ) -> None:
         """Handle the sensor state changes."""
-        new_state: State | None = event.data.get("new_state")
-        entity: str = event.data["entity_id"]
-
-        if (
-            new_state is None
-            or new_state.state is None
-            or new_state.state
-            in [
-                STATE_UNKNOWN,
-                STATE_UNAVAILABLE,
-            ]
-        ):
-            self.states[entity] = STATE_UNKNOWN
-            if not update_state:
-                return
-
-            self._calc_values()
-            self.async_write_ha_state()
-            return
-
-        if self._unit_of_measurement is None:
-            self._unit_of_measurement = new_state.attributes.get(
-                ATTR_UNIT_OF_MEASUREMENT
-            )
+        calc_states: dict[str, float | str] = {}
 
-        if self._unit_of_measurement != new_state.attributes.get(
-            ATTR_UNIT_OF_MEASUREMENT
-        ):
-            _LOGGER.warning(
-                "Units of measurement do not match for entity %s", self.entity_id
-            )
-            self._unit_of_measurement_mismatch = True
-
-        try:
-            self.states[entity] = float(new_state.state)
-            self.last = float(new_state.state)
-            self.last_entity_id = entity
-        except ValueError:
-            _LOGGER.warning(
-                "Unable to store state. Only numerical states are supported"
-            )
+        entity_state: State | None = None
+        entity: str | None = None
+        if event:
+            entity_state = event.data.get("new_state")
+            entity = event.data["entity_id"]
 
+        states: list[tuple[str, State]] = [
+            (entity_id, state)
+            for entity_id in self._entity_ids
+            if (state := self.hass.states.get(entity_id))
+        ]
+        for entity_id, state in states:
+            if (
+                state is None
+                or state.state is None
+                or state.state
+                in [
+                    STATE_UNKNOWN,
+                    STATE_UNAVAILABLE,
+                ]
+            ):
+                calc_states[entity_id] = STATE_UNKNOWN
+                if not update_state:
+                    self._calc_values(calc_states)
+                    return
```

Comment:
```suggestion
                continue
```

I still think a `continue` is needed here, because without it:
- The `min_max` sensor's initial state will be wrong if a sensor is missing, `unavailable` or `unknown`
- There will be a value error below, causing a log warning, which is not correct

Reply:
I intended to always log and thereby not having it. I understood now you meant to not log for these states